.github/workflows/iroha2-dev-pr.yml

```diff
@@ -16,67 +16,71 @@ env:
-        working-directory: actor
-        if: always()
-      - name: Verify wasm smartcontracts are able to build
-        run: mold --run cargo build --target wasm32-unknown-unknown --quiet
-        working-directory: wasm
-        env:
-          RUSTC_BOOTSTRAP: 1
       - name: Documentation check
+        if: always()
         run: |
           cargo doc --no-deps --quiet
           ./scripts/check.sh docs
-        if: always()
       - name: Check genesis
+        if: always()
         run: |
-           ./scripts/check.sh genesis
+          ./scripts/check.sh genesis
+      - name: Wasm build check
         if: always()
+        env:
+          RUSTC_BOOTSTRAP: 1
+        working-directory: wasm
+        run: mold --run cargo build --target wasm32-unknown-unknown --quiet
 
   # Coverage is both in PR and in push pipelines so that:
   # 1. PR can get coverage report from bot.
-  # 2. Coverage bot can have results from
-  # `iroha2-dev` to report coverage changes.
-  coverage:
+  # 2. Coverage bot can have results from `iroha2-dev` to report coverage changes.
+  tests_with_coverage:
     runs-on: ubuntu-latest
     container:
       image: 7272721/i2-ci:latest
     steps:
-      # TODO Remove this step #2165
-      - name: Add component
-        run: rustup component add llvm-tools-preview
-
+      # TODO Remove these steps #2165
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: llvm-tools-preview
+      - uses: taiki-e/install-action@cargo-llvm-cov
+      - uses: taiki-e/install-action@nextest
+      #
       - uses: actions/checkout@v2
       - uses: Swatinem/rust-cache@v1
-      - name: Run tests
-        run: |
-          mold --run cargo +nightly-2022-04-20 test \
-          --quiet --workspace --no-fail-fast -- \
-          --skip unstable_network --skip ui --test-threads 3
-        env:
-          RUSTFLAGS: "-C instrument-coverage"
-          LLVM_PROFILE_FILE: "iroha-%p-%m.profraw"
-      - name: Generate a grcov coverage report
-        run: |
-          grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/*" -o lcov.info
+      - name: Run tests and generate lcov report
         if: always()
+        run: mold --run cargo llvm-cov nextest --all-features --workspace --lcov --output-path lcov.info
       - name: Upload coverage to codecov.io
         uses: codecov/codecov-action@v2
-        if: always()
         with:
           file: lcov.info
+          fail_ci_if_error: true
+
+  integration_tests:
+    if: ${{ false }}
+    runs-on: ubuntu-latest
+    container:
+      image: 7272721/i2-ci:latest
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v2
+      - uses: Swatinem/rust-cache@v1
+      - name: Run tests
+        if: always()
+        run: |
+          mold --run cargo test --test '*' --all-features -- \
+          integration:: \
```

Comment:
`'*'` pattern matches any integration tests, then `integration::` limits them to client/tests/integration/
https://doc.rust-lang.org/cargo/commands/cargo-test.html#examples